module/Kbuild.in

```diff
@@ -1,31 +1,23 @@
 # When integrated in to a monolithic kernel the spl module must appear
 # first.  This ensures its module initialization function is run before
 # any of the other module initialization functions which depend on it.
-ZFS_MODULES += spl/
-ZFS_MODULES += avl/
-ZFS_MODULES += icp/
-ZFS_MODULES += lua/
-ZFS_MODULES += nvpair/
-ZFS_MODULES += unicode/
-ZFS_MODULES += zcommon/
-ZFS_MODULES += zfs/
-ZFS_MODULES += zstd/
-
-# The rest is only relevant when run by kbuild
-ifneq ($(KERNELRELEASE),)
-
-obj-$(CONFIG_ZFS) := $(ZFS_MODULES)
```

Comment:
The switch to using a `Kbuild` file and removing the `Makefile`'s has broken VPATH builds.  This is turn causes the `make rpm-kmod` target for the `generic/zfs-kmod.spec.in` to fail which depends on it.  For example:

```sh
sh autogen.sh 
mkdir  builddir
cd buiddir
../configure
make -s -j$(nproc)
...
Making all in module
Assembler messages:
Fatal error: can't create /g/g0/behlendo/src/git/zfs/builddir/module/os/linux/spl/.tmp_spl-atomic.o: No such file or directory
Assembler messages:
Fatal error: can't create /g/g0/behlendo/src/git/zfs/builddir/module/os/linux/spl/.tmp_spl-condvar.o: No such file or directory
```

We really should have a CI builder which does VPATH builds to catch things like this.  I happened to notice them when building packages locally.

Reply:
Just to be clear, by VPATH builds you mean out-of-tree ones like in the snippet? Because both fully out-of-tree and in tree/builddir like in the snippet (but I expect there to be no difference) work for me on bullseye, both now and at posting time.

Reply:
hm, if you're doing this on an ancient.. kernel? autohell? gmake? some combination of the five? it might not make the directories for the output files? but why? what's your platform?

Reply:
Yes, an out-of-tree build like in the snippet.  I happened to be testing with RHEL8, so a pretty old tool chain.  That could definitely be related.

Reply:
After an agonising day of pirating RHEL7 (not that it was my first choice, but RHEL8 straight up doesn't exist), I can thankfully repro this:
![image](https://user-images.githubusercontent.com/6709544/162084646-7b1b4bd6-eb9b-4793-ac79-ddbc15fd1711.png)

Reply:
Okay, I've added a fix for that on top

Reply:
Thanks!  I can confirm your latest commit did sort this out.  Out of curiosity were you unable to reproduce this with CentOS 7?  Just for reference you should be able to use CentOS Stream 8 as a good approximation for RHEL8.  It's effectively what will be the next RHEL minor release and is freely available.

Reply:
Didn't try CentOS at all, but I would've if I hadn't managed to configure for a kernel build without coming up with a repository licence. This is a great tip if I'm ever unfortunate enough to need it, though, thanks!

Reply:
If it is "incompletable", then CentOS Stream 8/9 will have to be good enough in that case.